graf2d/gpad/src/TPad.cxx

```diff
@@ -986,6 +986,7 @@ void TPad::Close(Option_t *)
 {
    if (!TestBit(kNotDeleted)) return;
    if (!fMother) return;
+   if (!fMother->TestBit(kNotDeleted)) return;
```

Comment:
This technique works most of the time but relies on undefined behavior.   It is for example likely to fails in the example:
```
{
   auto c = new TCanvas("canvas", "canvas"); // gPad=c;
   auto p = new TPad("pad", "pad", .2, .05, .8, .35 ); // fMother = gPad = c;
   delete c;
   auto other = new ClassSlightlySmallerThanTCanvas(....);
   p->Close(); 
}
```
where it might fails because the other objects is reusing the memory previously used by the TCanvas ...